### PR TITLE
[Testcase Refactoring]Decouple benchmark device handling from CUDA and XPU

### PR DESCRIPTION
**Pull request by Guang7u**
## Summary

This PR makes the Dynamo benchmark infrastructure device-generic by replacing hardcoded CUDA/XPU device handling with the accelerator and device-module APIs.

## Changes

 **benchmarks/dynamo/common.py**:

- Replaced CUDA/XPU-specific manual seed handling with the device-generic `torch.manual_seed` API.
- Added helpers to query the currently available accelerator and identify accelerator devices.
- Updated device cache cleanup to dispatch through `torch.get_device_module`.
- Generalized optimizer initialization from CUDA-only handling to the current accelerator type.
- Kept CUDA-specific Inductor accuracy workarounds restricted to CUDA devices.
- Updated default device selection to use the current available accelerator when CUDA is unavailable.
- Updated synchronization to use the synchronization API exposed by the selected device module.
- Addressed Claude review feedback on deterministic seeding by preserving the existing behavior that forces all `torch.manual_seed` calls to use seed 1337, while delegating device-specific seed handling to the device-generic `torch.random.manual_seed` implementation.
- Addressed Claude review feedback on optimizer compatibility by limiting `capturable=True` optimizer initialization to supported device types (`cuda`, `xpu`, `hpu`, `xla`, and PrivateUse1), preventing unsupported initialization on MPS.

**test/inductor/test_deterministic.py**:

- Replaced `--backend inductor` with `--inductor` in the run-to-run determinism test so that the complete Inductor benchmark configuration is enabled.
- The generic `--backend inductor` path invokes `torch._dynamo.optimize` directly, bypassing the wrapper patched by `torch_npu`. Using the dedicated `--inductor` option follows the standard Inductor benchmark path and ensures that the `torch_npu` wrapper is applied correctly.

## Test plan

```bash
python test/inductor/test_deterministic.py
```
Verified locally that the relevant benchmark cases run correctly on CUDA and NPU.

## Notes

This is part of the device-generic benchmark refactoring initiative tracked in [Test] PyTorch Test Case Refactoring and Track [#185590](https://github.com/pytorch/pytorch/issues/185590).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -1869,12 +1866,27 @@ def setup_amp(self, current_device=None):
...
+            device_type in capturable_devices
+            and self.args.training
+            and name not in CI_SKIP_OPTIMIZER
+        ):
             use_sgd = (
                 (name in CI_USE_SGD and self.args.ci)
                 or name in BENCHMARK_USE_SGD
                 or (
-                    torch.version.hip is None
+                    device_type == "cuda"
```
Comment (FuDdd) [inline benchmarks/dynamo/common.py:1888]: What's this device_type == "cuda" used for?

**Review by FuDdd — APPROVED** [review]

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (Guang7u) [reply benchmarks/dynamo/common.py:1888]: device_type == "cuda" has removed

**Comment by Guang7u** [comment]
@jansel  , could you please review this PR again? I've made the corresponding changes based on Claude's feedback. Below are the changes I made.

1. Addressed Claude review feedback on deterministic seeding by preserving the existing behavior that forces all torch.manual_seed calls to use seed 1337, while delegating device-specific seed handling to the device-generic torch.random.manual_seed implementation.

2. Addressed Claude review feedback on optimizer compatibility by limiting capturable=True optimizer initialization to supported device types (cuda, xpu, hpu, xla, and PrivateUse1), preventing unsupported initialization on MPS.

**Comment by jansel** [comment]
@claude review again
